src/test/java/com/samsolutions/employeesdep/repository/DepartmentRepositoryTest.java

```diff
@@ -0,0 +1,64 @@
+package com.samsolutions.employeesdep.repository;
+
+import com.samsolutions.employeesdep.model.entities.Department;
+import com.samsolutions.employeesdep.model.repository.DepartmentRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.test.annotation.Rollback;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SpringBootTest
+public class DepartmentRepositoryTest {
+    @Autowired
+    private DepartmentRepository repository;
+
+    private Department depart1;
+    private Department depart2;
+
+    @BeforeEach
+    public void setup() {
+        depart1 = new Department("Java Department");
+        repository.save(depart1);
+        assertEquals("Java Department", depart1.getName());
+        depart2 = new Department(".NET Department");
+        repository.save(depart2);
+        assertEquals(".NET Department", depart2.getName());
+    }
+
+    @Test
+    void testFindAll() {
+        List<Department> departs = repository.findAll();
+        assertEquals(2, departs.size());
+        assertNotEquals(depart1.getId(), depart2.getId());
+    }
+
+    @Test
+    void testGetByName() {
+        Department depart = repository.findByName("Java Department");
+        assertEquals("Java Department", depart.getName());
+    }
+
+    @Test
+    void testUpdate() {
+        depart1.setName("PHP Department");
+        repository.save(depart1);
+        assertEquals("PHP Department", depart1.getName());
+        assertNotEquals("Java Department", depart1.getName());
+    }
+
+    @AfterEach
+    public void tearDown() {
+        repository.deleteById(depart1.getId());
+        repository.delete(depart2);
+        assertEquals(0, repository.findAll().size());
```

Comment:
assertThat(repository.findAll(), is(empty()));

Reply:
done